cmd/autoconfigbrancher/main.go

```diff
@@ -0,0 +1,170 @@
+	flag.Var(&o.futureReleases, "future-release", "Configurations will get branched to target this release, provide one or more times.")
+	flag.Parse()
+	return o
+}
+
+func validateOptions(o options) error {
+	if o.githubLogin == "" {
+		return fmt.Errorf("--github-login cannot be empty string")
+	}
+	if o.githubToken == "" {
+		return fmt.Errorf("--github-token is mandatory")
+	}
+	if (o.gitEmail == "") != (o.gitName == "") {
+		return fmt.Errorf("--git-name and --git-email must be specified together")
+	}
+	if o.targetDir == "" {
+		return fmt.Errorf("--target-dir is mandatory")
+	}
+	if o.assign == "" {
+		return fmt.Errorf("--assign is mandatory")
+	}
+	if o.currentRelease == "" {
+		return fmt.Errorf("--current-release is mandatory")
+	}
+	if len(o.futureReleases.Strings()) == 0 {
+		return fmt.Errorf("--future-release is mandatory")
+	}
+	return nil
+}
+
+func hasNewCommits() (bool, error) {
+	cmd := "git"
+	args := []string{"rev-list", "--count", "HEAD", "^before"}
+	logrus.WithField("cmd", cmd).WithField("args", args).Info("running command ...")
+	stdout, _, err := helper.Call(cmd, args...)
+	if err != nil {
+		return false, err
+	}
+	i, err := strconv.Atoi(strings.TrimSuffix(string(stdout), "\n"))
+	if err != nil {
+		return false, err
+	}
+	return i > 0, nil
+}
+
+func run(cmd string, args ...string) {
+	logrus.WithField("cmd", cmd).WithField("args", args).Info("running command ...")
+	if _, stderr, err := helper.Call(cmd, args...); err != nil {
+		logrus.WithError(err).WithField("string(stderr)", string(stderr)).Fatal("Failed to run command")
+	}
+}
+
+func main() {
+	o := parseOptions()
+	if err := validateOptions(o); err != nil {
+		logrus.WithError(err).Fatal("Invalid arguments.")
+	}
+
+	sa := &secret.Agent{}
+	if err := sa.Start([]string{o.githubToken}); err != nil {
+		logrus.WithError(err).Fatal("Failed to start secrets agent")
+	}
+
+	gc := github.NewClient(sa.GetTokenGenerator(o.githubToken), sa.Censor, github.DefaultGraphQLEndpoint, github.DefaultAPIEndpoint)
+
+	logrus.Infof("Changing working directory to '%s' ...", o.targetDir)
+	if err := os.Chdir(o.targetDir); err != nil {
+		logrus.WithError(err).Fatal("Failed to change to root dir")
+	}
+
+	cmd := "git"
+	args := []string{"tag", "before"}
+	run(cmd, args...)
+
+	cmd = "/usr/bin/determinize-ci-operator"
+	args = []string{"--config-dir", "./ci-operator/config", "--current-release", o.currentRelease}
+	for _, fr := range o.futureReleases.Strings() {
+		args = append(args, []string{"--future-release", fr}...)
+	}
+	args = append(args, "--confirm")
+	run(cmd, args...)
+
+	cmd = "/usr/bin/config-brancher"
+	run(cmd, args...)
+
+	cmd = "/usr/bin/ci-operator-prowgen"
+	args = []string{"--from-dir", "./ci-operator/config", "--to-dir", "./ci-operator/jobs"}
+	run(cmd, args...)
+
+	// Here we use `git-rev-list`, together with the tag `before` to check
+	// if any new commits has been generated so far by the release-tools above.
```

Comment:
What is actually making commits? I don't see it anywhere

Reply:
I thought the 3-tool calls generate commits.
I cannot test it on master of release repo today because it does not generate any change.

Reply:
No, the tools just change files, they do not do any changes to `git`

Reply:
Then I misunderstood your comment from JIRA card.
Working on this now.

Reply:
Done.